pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -31,15 +34,29 @@
 
   private Predicate predicate;
   private int nextBlockCallCounter = 0;
+  private PredicateEvaluator _predicateEvaluator;
 
   public void setPredicate(Predicate predicate) {
     this.predicate = predicate;
   }
 
+  public void setPredicateEvaluator(PredicateEvaluator predicateEvaluator) {
```

Comment:
Removed all setPredicate* from the BaseFilterOperator class. Constructor of concrete filter operators now take the predicate.